addons/mail/models/mail_channel.py

```diff
@@ -228,12 +228,7 @@ def create(self, vals):
 
     def unlink(self):
         # Delete mail.channel
-        admin_channel = False
-        for channel in self:
-            if channel.get_external_id()[channel.id] == 'mail.channel_admins':
-                admin_channel = channel
-        if admin_channel:
-            self -= admin_channel
+        if any(channel_xmlid == 'mail.channel_admins' for channel_xmlid in self.get_external_id().values()):
```

Comment:
fixup in previous commit

Reply:
Yes the last commit is a temporary one with some last updates & fixes that will be squashed in commit history after review *.